core/modules/simpletest/simpletest.module

```diff
@@ -411,8 +424,17 @@ function simpletest_test_get_all() {
           }
         }
         if (!empty($error)) {
-          watchdog('simpletest', $error, array('%uri' => $file->uri), WATCHDOG_WARNING, l(t('See the Testing Framework documentation.'), 'https://docs.backdropcms.org/documentation/testing-framework'));
-          continue;
+          if (backdrop_is_cli()) {
+            // If running in CLI, then we're probably about to run some
+            // automated tests, so we need a hard fail.
+            print format_string($error, $variables) . "\n";
+            exit(1);
+          }
+          else {
+            backdrop_set_message(format_string($error, $variables), 'error');
```

Comment:
As I've elaborated in detail in my comment in the issue, this is not proper use of `t()` AFAIK. `$error` should remain a plain string (non-`t()`'ed), with the substitution placeholders, and then you should be doing this here:

```suggestion
            backdrop_set_message(t($error, $variables), 'error');
```

Reply:
...but if you already `t()` `$error` as in my suggestions, then you can simply do this here:
```suggestion
            backdrop_set_message($error, 'error');
```